source/particles/particle_handler.cc

```diff
@@ -1356,41 +1362,54 @@ namespace Particles
           // previous cell Most likely we will find the particle in them.
           for (unsigned int i = 0; i < n_neighbor_cells; ++i)
             {
-              try
+              typename std::set<typename Triangulation<dim, spacedim>::
+                                  active_cell_iterator>::const_iterator cell =
+                vertex_to_cells[closest_vertex_index].begin();
+
+              std::advance(cell, neighbor_permutation[i]);
+              mapping->transform_points_real_to_unit_cell(*cell,
+                                                          real_locations,
+                                                          reference_locations);
```

Comment:
While not related to this PR, this code reminds me that I would like to try (at some point in the future) if we can somehow group *nearby* points that left their cell, and send all those *nearby* points into the same `transform_points_real_to_unit_cell` call. In the case of many particles per cell, the will be a good chunk of particles traveling between the same two cells, so if we only got the grouping right we might get up to 4x speedup for 4-wide vectorization (AVX) for free, and for expensive manifolds we would get even more.